src/test/java/io/confluent/kafkarest/unit/ConsumerManagerTest.java

```diff
@@ -172,38 +177,50 @@ public void testConsumerNormalOps() throws InterruptedException, ExecutionExcept
     String cid = consumerManager.createConsumer(
         groupName, new ConsumerInstanceConfig(EmbeddedFormat.BINARY));
     sawCallback = false;
+    actualException = null;
```

Comment:
@gawth I've also used a special test base class to handle background exceptions. See https://github.com/apache/kafka/blob/trunk/connect/runtime/src/test/java/org/apache/kafka/connect/util/ThreadedTest.java for example. Would this approach work too so you could just do things like `assertNull(e)` in some of the callbacks and still get them propagated to the main test? The obvious drawback is that the failure becomes a bit disconnected from the origin of the error, but that's true in both cases.

I'm not sure if it is worth the change here since these cases need to record `actualRecords` too, unless we can just keep all assertions in the event handlers if we propagate exceptions back to the main test correctly. What do you think?